pypesto/sample/geweke_test.py

```diff
@@ -46,7 +46,11 @@ def spectrum(x: np.ndarray, nfft: int = None, nw: int = None) -> np.ndarray:
         n = nw
 
     # Number of windows
-    k = np.floor((n - n_overlap) / (nw - n_overlap)).astype(int)
+    k = (
+        np.floor((n - n_overlap) / (nw - n_overlap)).astype(int)
+        if nw != n_overlap
```

Comment:
When is `nw == n_overlap`? Is the total sample size < 100? If so, then I guess this could be handled earlier, since it looks like a vector of zeroes will be returned if `k == 0`. Fine I guess but, if this is an estimate of the variance, then it probably breaks the z-score calculation at line 158.

Reply:
> When is `nw == n_overlap`?

* If `len(x) <= 3` and `nfft <= 3` and `nw == None`, for the current usages in pypesto, the first condition is sufficient
* If `len(x) <= 3` and `nfft == None` and `nw == None`
* If `nw < 2`

> this could be handled earlier

Yes.

I don't want to touch that more than necessary right now, but if somebody else feels like cleaning that up (or replacing it by the scipy implementation right away), there is certainly room for improvement.

I would merge this change rather soon, since it at least solves the ridiculously long looping and maintains the old behavior as experienced on most systems.

> Fine I guess but, if this is an estimate of the variance, then it probably breaks the z-score calculation at line 158.

That would have been the case since 2020 then. Would be great if somebody could take a deeper look.

Reply:
> looks like a vector of zeroes will be returned if `k == 0`. Fine I guess but, if this is an estimate of the variance, then it probably breaks the z-score calculation at line 158.

From everything I've seen, it either returns `[]`, which is just skipped in `spectrum0`, or it returns `[nan]`, which will make `calculate_zscore` return `nan`.